### PR TITLE
ratelimiter: remove anonymous attribute to keep threadsafe

### LINKED ISSUE #5037: pkg/ratelimit/ratelimiter: the function `RateLimiter.Available`  is not threadsafe.
## Enhancement Task
It seems we try to use `mu` to make the function `Available` threadsafe, however, the caller can still call`RateLimiter.ReserveN(XXX)` concurrently since `rate.Limiter` was an anonymous attribute for `Ratelimiter` 
[https://github.com/tikv/pd/blob/04d6881d20b367542da4c924685931cbce1c9cae/pkg/ratelimit/ratelimiter.go#L27-L60](https://github.com/tikv/pd/blob/04d6881d20b367542da4c924685931cbce1c9cae/pkg/ratelimit/ratelimiter.go#L27-L60)

### PR DESCRIPTION
**Pull request by CabinfeverB**
Signed-off-by: Cabinfever_B <cabinfeveroier@gmail.com>

### What problem does this PR solve?

Issue Number: Close #5037

### What is changed and how does it work?
make a name for`rate.Limiter` for removing anonymous attribute 

```commit-message
remove anonymous attribute to keep threadsafe
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by AndreMouche — COMMENTED** [review]
please fix the CI ~~

**Review by AndreMouche — COMMENTED** [review]
LGTM

File: pkg/ratelimit/ratelimiter.go [hunk 1]
```diff
@@ -57,5 +58,38 @@ func (l *RateLimiter) AllowN(n int) bool {
 	l.mu.Lock()
 	defer l.mu.Unlock()
 	now := time.Now()
-	return l.Limiter.AllowN(now, n)
+	return l.limiter.AllowN(now, n)
+}
+
+// SetBurst is shorthand for SetBurstAt(time.Now(), newBurst).
+func (l *RateLimiter) SetBurst(burst int) {
```
Comment (AndreMouche) [inline pkg/ratelimit/ratelimiter.go:65]: I think we still need lock to make it threadsafe..

**Review by rleungx — APPROVED** [review]
LGTM, except for the unresolved comment.

**Review by AndreMouche — APPROVED** [review]
LGTM

**Pull request merged**